### PR TITLE
Apply more automatic Java refactoring

### PR DESCRIPTION
**Pull request by Baltoli**
This PR applies a few small simplifications suggested by IntelliJ:
* [Expression lambdas](https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html)
* Removal of dead-code default cases in switches
* Simplification of stream + optional call chains (removal of redundant calls etc.)

### CODE REVIEW

File: kernel/src/main/java/org/kframework/compile/ConvertDataStructureToLookup.java [hunk 1]
```diff
@@ -121,7 +121,6 @@ public static Set<KLabel> filteredMapConstructors(Module m) {
     return stream(m.productions())
         .filter(p -> p.att().contains(Att.ASSOC()) && p.att().contains(Att.FILTER_ELEMENT()))
         .map(p -> p.klabel().get())
-        .distinct()
         .collect(Collectors.toSet());
```
Comment (radumereuta) [inline kernel/src/main/java/org/kframework/compile/ConvertDataStructureToLookup.java:124]: Isn't there a `.toSet()` variant to get rid of this extra line?

**Review by radumereuta — APPROVED** [review]
lgtm, just one question

File: kernel/src/main/java/org/kframework/compile/ConvertDataStructureToLookup.java (hunk 1 above)
Comment (Baltoli) [reply kernel/src/main/java/org/kframework/compile/ConvertDataStructureToLookup.java:124]: Weirdly, no, despite the existence of `.toList()`!

**Review by Robertorosmaninho — APPROVED** [review]
LGTM! Thanks for working on this!

**Pull request merged**